services/userService.js

```diff
@@ -0,0 +1,139 @@
+const jwt = require('jsonwebtoken');
+const requestPromise = require('request-promise');
+const userDao = require('../models/userDao');
+
+const signIn = async ( code ) => {
+  const apiKey = process.env.API_KEY;
+  const redirectUri = process.env.REDIRECT_URI;
+
+  const getKakaoTokenOption = {
+    uri: "https://kauth.kakao.com/oauth/token",
+    method: "POST",
+    form: {
+      grant_type: "authorization_code",
+      client_id: apiKey,
+      redirect_uri: redirectUri,
+      code: code
+    },
+    headers: {
+        "content-type" : "application/x-www-form-urlencoded"
+    },
+    json: true
+  }
+
+  const getKakaoToken = await requestPromise(getKakaoTokenOption);
+  const KAKAO_TOKEN = getKakaoToken['access_token'];
+
+  const getKakaoUserOption = {
+    uri: "https://kapi.kakao.com/v2/user/me",
+    method: "POST",
+    headers: {
+        "content-type" : "application/x-www-form-urlencoded",
+        "Authorization" : `Bearer ${KAKAO_TOKEN}`
+    },
+    json: true
+  }
+
+  const kakaoUserInfo = await requestPromise(getKakaoUserOption);
+
+  if ( !kakaoUserInfo.id ) {
+    const err = new Error('KAKAO_ID_IS_NOT_VALID');
+    err.statusCode = 400;
+    throw err;
+  }
+
+  const checkUserBySocialId = await userDao.getUserBySocialId( kakaoUserInfo.id );
+  
+  if ( !checkUserBySocialId || checkUserBySocialId.length === 0 ) {
+
+    if ( !kakaoUserInfo.kakao_account.email ) {
+      kakaoUserInfo.kakao_account.email = "";
+    }
+
+    const kakaoUserData = {
+      id : kakaoUserInfo.id,
+      email : kakaoUserInfo.kakao_account.email
+    }
+
+    return kakaoUserData;
+  }
```

Comment:
이중 조건문의 조건을 하나로 합쳐 가독성을 높일 수 있을 것 같습니다!
```suggestion
  const KAKAO_EMAIL = kakaoUserInfo.kakao_account.email
  if ( !user && !KAKAO_EMAIL) {
    const data = {
      id : kakaoUserInfo.id,
      email : KAKAO_EMAIL
    }

    return data;
  }
```